File: frontend/erp-portfolio-web/src/types/categoria.ts
/**
 * =====================================================================
 * Arquivo....: categoria.ts
 * Versão.....: 1.0.0
 * Data.......: 21/09/2026
 * Descrição..: Tipos do cadastro de Categorias, espelhando os DTOs da API
 *              (CategoriaRespostaDto, CategoriaCriacaoDto,
 *              CategoriaAtualizacaoDto e CategoriaFiltroDto).
 * ---------------------------------------------------------------------
 * Histórico de alterações:
 *   1.0.0 - 21/09/2026 - Criação do arquivo.
 * =====================================================================
 */

export interface Categoria {
  id: number
  nome: string
  ativo: boolean
  /** Data/hora ISO 8601 em UTC. */
  dataCadastro: string
}

export interface CategoriaCriacao {
  nome: string
}

export interface CategoriaAtualizacao extends CategoriaCriacao {
  ativo: boolean
}

export interface CategoriaFiltro {
  /** Trecho do nome. */
  busca?: string
  /** true = só ativas, false = só inativas, ausente = todas. */
  ativo?: boolean
  pagina: number
  tamanhoPagina: number
}

File: backend/ErpPortfolio.Api/Models/Categoria.cs
// =====================================================================================
// Arquivo....: Categoria.cs
// Versão.....: 1.0.0
// Data.......: 21/09/2026
// Descrição..: Entidade de domínio que representa uma categoria de produtos.
// -------------------------------------------------------------------------------------
// Banco......: PostgreSQL - erp_portfolio_db
// Tabelas....: public.categorias
// Fontes.....: Mapeada em ErpPortfolioDbContext.Categorias (EF Core / Npgsql).
//              Colunas: id, nome, ativo, data_cadastro (o mapeamento de nomes fica no
//              DbContext).
// -------------------------------------------------------------------------------------
// Histórico de alterações:
//   1.0.0 - 21/09/2026 - Criação do arquivo.
// =====================================================================================

namespace ErpPortfolio.Api.Models;

public class Categoria
{
    public int Id { get; set; }

    /// <summary>Nome único, sem diferenciar maiúsculas/minúsculas (a checagem fica no serviço).</summary>
    public string Nome { get; set; } = string.Empty;

    public bool Ativo { get; set; } = true;

    /// <summary>Data/hora de cadastro em UTC.</summary>
    public DateTime DataCadastro { get; set; }
}
